frontend/src/queries/fileQueries.ts

```diff
@@ -0,0 +1,463 @@
+      throw new Error('No file share path selected');
+    }
+
+    const url = getFileBrowsePath(fspName, folderName);
+
+    const response = await sendFetchRequest(url, 'GET', undefined, { signal });
+    const body = await response.json();
+
+    if (response.ok) {
+      return body as FileBrowserResponse;
+    }
+
+    // Handle error responses
+    if (response.status === 403) {
+      const errorMessage =
+        body.info && body.info.owner
+          ? `You do not have permission to list this folder. Contact the owner (${body.info.owner}) for access.`
+          : 'You do not have permission to list this folder. Contact the owner for access.';
+      // Include partial data (info) if available from backend
+      const error = new FetchError(
+        response,
+        errorMessage,
+        body.info ? { info: body.info } : undefined
+      );
+      throw error;
+    } else if (response.status === 404) {
+      throw new Error('Folder not found');
+    } else {
+      throw new Error(
+        body.error ? body.error : `Unknown error (${response.status})`
+      );
+    }
+  };
+
+  const transformData = React.useCallback(
+    (data: FileBrowserResponse | { info: FileOrFolder }): FileQueryData => {
+      // This should never happen because query is disabled when !fspName
+      if (!fspName) {
+        throw new Error('fspName is required for transforming file query data');
+      }
+
+      const fspKey = makeMapKey('fsp', fspName);
+      const currentFileSharePath =
+        (zonesAndFspQuery.data?.[fspKey] as FileSharePath) || null;
+
+      // Normalize the path in the current file or folder
+      let currentFileOrFolder: FileOrFolder | null = data.info;
+      if (currentFileOrFolder) {
+        currentFileOrFolder = {
+          ...currentFileOrFolder,
+          path: normalizePosixStylePath(currentFileOrFolder.path)
+        };
+      }
+
+      // Normalize file paths and sort: directories first, then alphabetically
+      // Handle partial data case (403 error with only info, no files)
+      const rawFiles = 'files' in data ? data.files : [];
+      let files = (rawFiles || []).map(file => ({
+        ...file,
+        path: normalizePosixStylePath(file.path)
+      })) as FileOrFolder[];
+
+      files = files.sort((a: FileOrFolder, b: FileOrFolder) => {
+        if (a.is_dir === b.is_dir) {
+          return a.name.localeCompare(b.name);
+        }
+        return a.is_dir ? -1 : 1;
+      });
+
+      return {
+        currentFileSharePath,
+        currentFileOrFolder,
+        files
+      };
+    },
+    [fspName, zonesAndFspQuery.data]
+  );
+
+  const query = useQuery<FileBrowserResponse, Error, FileQueryData>({
+    queryKey: fileQueryKeys.filePath(fspName || '', folderName),
+    queryFn: fetchFileInfo,
+    select: (data: FileBrowserResponse) => transformData(data),
+    enabled: !!fspName && !!zonesAndFspQuery.data,
+    staleTime: 5 * 60 * 1000 // 5 minutes - file listings don't change that often
```

Comment:
This is an interesting choice and maybe a good one (since you can still get updates with the Refresh button), but  let's discuss this some more at the next meeting. I'm also interested in all the other staleTime settings, and if the Refresh workaround works in all cases.